src/main/java/io/jenkins/plugins/folderauth/roles/AbstractRole.java

```diff
@@ -43,6 +44,27 @@ public AbstractRole(String name, Set<PermissionWrapper> permissionWrappers) {
         this.permissionWrappers = Collections.unmodifiableSet(permissionWrappers);
     }
 
+    /**
+     * Constructor for an {@link AbstractRole} that does not use thread-safe constructs.
+     * <p>
+     * Use only when serializing the object. The set of Sids and the unmodifiable set of
+     * permissions is changed to a {@link HashSet} for simpler serialization
+     *
+     * @param name        the names of role
+     * @param permissions the permissions granted by this role
+     * @param sids        the sids to be assigned to this role
+     */
+    @ParametersAreNonnullByDefault
```

Comment:
To save your time, you can apply it on a package level: https://stackoverflow.com/questions/5516157/how-do-i-get-parametersarenonnullbydefault-to-work